### PR TITLE
feat(TPG>=4.81.0)!: add fqdn policies

### PR DESCRIPTION
**Pull request by TheKangaroo**
Fixes https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1728
Please let me know if anything is missing. This is my first PR to this repo.

### CODE REVIEW

**Comment by TheKangaroo** [comment]
@Jberlinsky @ericyz I fixed a bug in the template and the lint should work now. At least it worked on my local machine.

**Comment by ericyz** [comment]
Hi @TheKangaroo - thank you for your contribution. I'll review and trigger the CI

**Comment by ericyz** [comment]
A note for myself. The feature introduced in
https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.81.0

**Review by ericyz — APPROVED** [review]

**Comment by ericyz** [comment]
LGTM

**Review by shakeebshams — APPROVED** [review]
LGTM

**Comment by TheKangaroo** [comment]
I merged the latest changes from master, because of version conflicts. I hope this is ok, I'm not too familiar with the PR process in this repo.

**Comment by ericyz** [comment]
@TheKangaroo - That's all good. Once the Integration test passes, we'll merge

File: README.md [hunk 1]
```diff
@@ -128,132 +128,6 @@ Then perform the following commands on the root folder:
 - `terraform destroy` to destroy the built infrastructure
 
 <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
-## Inputs
```
Comment (apeabody) [inline README.md:131]: Thanks for the pull request @TheKangaroo.

The Readme(s) appear to have missing content.  Can you try running `make docker_generate_docs`?

Comment (TheKangaroo) [reply README.md]: sorry, should be fixed now

**Comment by TheKangaroo** [comment]
Is there anything I need to fix to make the pipeline succeed? I think I cannot see the pipeline output.

**Comment by apeabody** [comment]
> Is there anything I need to fix to make the pipeline succeed? I think I cannot see the pipeline output.

I re-triggered it as the failure didn't appear related to the code change.

**Comment by TheKangaroo** [comment]
Hi, I don't want to bother you but is there anything I can do to get the PR merged soon? We have 'manually' enabled the feature on some clusters and now we can't update the provider until we can set the option via the module.

**Comment by apeabody** [comment]
> Hi, I don't want to bother you but is there anything I can do to get the PR merged soon? We have 'manually' enabled the feature on some clusters and now we can't update the provider until we can set the option via the module.

Trigger a fresh run of the CI

File: autogen/main/variables.tf.tmpl [hunk 2]
```diff
@@ -559,6 +559,12 @@ variable "workload_config_audit_mode" {
   type        = string
   default     = "DISABLED"
 }
+
+variable "enable_fqdn_network_policy" {
+  type        = bool
+  description = "Enable FQDN Network Policies on the cluster"
+  default     = false
```
Comment (apeabody) [inline autogen/main/variables.tf.tmpl:566]: As this is optional, would `null` be a better default? https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#enable_fqdn_network_policy

**Review by apeabody — DISMISSED** [review]
Thanks for the contribution @TheKangaroo!

**Comment by TheKangaroo** [comment]
@Jberlinsky @apeabody Can we continue with this PR?

**Comment by apeabody** [comment]
Thanks for the contribution @TheKangaroo!

**Pull request merged**